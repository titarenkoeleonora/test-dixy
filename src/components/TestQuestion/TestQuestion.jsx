import React from 'react';

const TestQuestion = (props) => {
  const {question} = props;
  const {title, answers, id} = question;

  return (
    <div>
      <p className="form-test__question">{title}</p>
      <ul className="form-test__answers-list">
        {answers.map((answer, index) => (
          <li className="form-test__answers-item" key={`${index}-${answer.answer}`}>
            <input className="form-test__answer-input checkbox visually-hidden" type="radio" name="question-list" id={id} value={answer.variant} required/>
            <label className="form-test__answer-text" htmlFor={id}>{answer.answer}</label>
          </li>
        ))}
      </ul>
      <button className="form-test__button" type="submit" aria-label="Далее">Далее</button>
    </div>
  );
};

export default TestQuestion;