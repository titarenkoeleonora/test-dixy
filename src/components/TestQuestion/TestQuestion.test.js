import TestQuestion from './TestQuestion';
import renderer from 'react-test-renderer';

const testQuestion = {
  id: '01',
  title: 'СПИСОК ПОКУПОК – ЭТО…',
  answers: [{
    answer: 'Список чего? У меня все есть.',
    variant: 'a',
  }], 
};

it(`Correctly TestQuestion render`, () => {
  const tree = renderer
    .create(
      <TestQuestion
        question={testQuestion}
      />
    );

  expect(tree).toMatchSnapshot();
});
