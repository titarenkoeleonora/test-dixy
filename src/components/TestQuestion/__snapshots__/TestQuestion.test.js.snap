// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly TestQuestion render 1`] = `
<div>
  <p
    className="form-test__question"
  >
    СПИСОК ПОКУПОК – ЭТО…
  </p>
  <ul
    className="form-test__answers-list"
  >
    <li
      className="form-test__answers-item"
    >
      <input
        className="form-test__answer-input checkbox visually-hidden"
        id="question-list-0"
        name="question-list"
        type="radio"
        value="a"
      />
      <label
        className="form-test__answer-text"
        htmlFor="question-list-0"
      >
        Список чего? У меня все есть.
      </label>
    </li>
  </ul>
  <button
    aria-label="Далее"
    className="form-test__button"
    type="submit"
  >
    Далее
  </button>
</div>
`;
